tests/conftest.py

```diff
@@ -75,13 +89,35 @@ async def lifespan(app: FastAPI) -> AsyncIterator[dict[str, Any]]:
 
 
 @pytest.fixture
-def empty_stapi_client(base_url: str) -> Iterator[TestClient]:
+def stapi_client_async_opportunity(
+    mock_products: list[Product],
+    base_url: str,
+    mock_opportunities: list[Opportunity],
+) -> Iterator[TestClient]:
```

Comment:
I think the type is supposed to be `Generator[TestClient, None, None]` instead of Iterator

Reply:
[mypy guidance](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#generators) seems OK with using `Iterator`. But I have no issue with being more explicit. Updated.